devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/extension/QuarkusPluginExtension.java

```diff
@@ -182,4 +188,13 @@ public Path appJarOrClasses() {
         }
         return classesDir;
     }
+
+    public Map<String, String> getQuarkusBuildProperties() {
+        return quarkusBuildProperties;
+    }
+
+    public void set(String name, @Nullable String value) {
+        quarkusBuildProperties.put(String.format("quarkus.%s", name), value);
+    }
```

Comment:
This should use [lazy configuration APIs](https://docs.gradle.org/current/userguide/lazy_configuration.html). So `quarkusBuildProperties` should not be of type `Map<String, String>` but `MapProperty<String, String>`. At the same time there should be an overload for `set` where value can be passed as a `Property<String>`.

Another issue with this solution is that is assumes everything is a string, but that's not the case. For example the `package.output-directoy` should probably be of type `DirectoryProperty` (see https://docs.gradle.org/current/userguide/lazy_configuration.html#working_with_files_in_lazy_properties).

Reply:
@britter thanks a lot for the review. Let us know if you would like to submit a PR yourself. That'd be much appreciated.

Reply:
Thank you for the feedback @britter , I'm reviewing the docs. I can also submit a PR to fix this, if you won't.

Reply:
I don't see myself doing this before the end of the month. So if you have some time @jacobdotcosta please go ahead.

Reply:
It doesn't look like it was fixed yet. @britter in case you have some time to contribute the fixes you mentioned previously, it'll be much appreciated. Thanks!

Reply:
@aloubyansky I managed to address the first part of it in #30713. I'm not sure it's possible to address the second part using a map based approached.